### PR TITLE
bugfix: RM did not clear XID after the local transaction threw an exception

### LINKED ISSUE #4020: RM执行本地事务抛出异常之后，没有执行清除xid，导致线程一直绑定着错误请求的xid
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
AT模式
RM执行本地事务操作时出现异常，发现 RootContext的CONTEXT_HOLDER内线程对应的XID没有被清除，
跟踪发现没有执行以下两个地方的清除xid代码：
· io.seata.integration.http.TransactionPropagationInterceptor#postHandle
· io.seata.integration.http.HttpHandlerExceptionResolver#doResolveException

导致线程池内的线程处理完成请求之后回收到池内，但是依然在RootContext的CONTEXT_HOLDER关联了一个XID，出现的问题如下：
新的GlobalTransaction进来时执行io.seata.integration.http.TransactionPropagationInterceptor#preHandle方法的如下代码时：
`

    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
        String xid = RootContext.getXID();
        String rpcXid = request.getHeader(RootContext.KEY_XID);

        if (LOGGER.isDebugEnabled()) {
            LOGGER.debug("xid in RootContext[{}] xid in HttpContext[{}]", xid, rpcXid);
        }
        if (**xid == null && rpcXid != null**) {
            RootContext.bind(rpcXid);
            if (LOGGER.isDebugEnabled()) {
                LOGGER.debug("bind[{}] to RootContext", rpcXid);
            }
        }

        return true;
    }
`
当上次出现异常的线程再次从Pool被分配来处理一个新的@GlobalTransaction请求时，由于从RootContext内已经获取得到xid，
按 TransactionPropagationInterceptor#preHandle 的逻辑，这时候会忽略从Header带进来的xid（request.getHeader(RootContext.KEY_XID)）  带着Context内绑定的旧xid执行本地事务，当执行完成需要注册分支事务时，由于xid是旧的，会导致抛出这个错误：
`
Could not found global transaction xid = （旧xid）, may be has finished
`

我们项目内有 @RestControllerAdvice 处理的全局异常处理（但是处理时正常返回Response 并未Throw Exception）：
`

 @RestControllerAdvice
 public class LepuRestExceptionHandler {

    @ExceptionHandler(Throwable.class)
    @ResponseStatus(HttpStatus.OK)
    public R handleError(Throwable e) {
        log.error("服务器异常", e);
        return R.fail(ResultCode.INTERFACE_INNER_INVOKE_ERROR, ResultCode.INTERFACE_INNER_INVOKE_ERROR.getMessage());
    }

`

注：我在Isses内有发现不少 Could not found global transaction xid = xxx, may be has finished 的问题，但是没有提到我上述的这种场景，是否是我这边的用法不对？

经过测试 发现异常走了 @RestControllerAdvice 处理之后转换成普通的Result，就到不了HttpHandlerExceptionResolver。

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version : 1.8
- Seata version:   spring-boot-seata-starter  version 1.4.2
- OS : Linux
- Others: SpringBoot 2.3.8

### PR DESCRIPTION
**Pull request by lvekee**
### Ⅰ. Describe what this PR did
bugfix:RM fails to clear XID after throwing an exception after executing local transactions, resulting in the thread always binding the XID of the wrong request

### Ⅱ. Does this pull request fix one issue?
fixes #4020.

### CODE REVIEW

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by caohdgege — APPROVED** [review]
LGTM

**Review by slievrly — COMMENTED** [review]
LGTM

File: integration/http/src/main/java/io/seata/integration/http/XidResource.java [hunk 1]
```diff
@@ -32,14 +32,14 @@
...
-        if (xid != null) {
+        if (StringUtils.isNotBlank(xid)) {
             String unbindXid = RootContext.unbind();
             if (LOGGER.isDebugEnabled()) {
                 LOGGER.debug("unbind[{}] from RootContext", unbindXid);
             }
             if (!StringUtils.equalsIgnoreCase(rpcXid, unbindXid)) {
                 LOGGER.warn("xid in change during RPC from {} to {}", rpcXid, unbindXid);
-                if (unbindXid != null) {
+                if (StringUtils.isNotBlank(unbindXid)) {
```
Comment (wangliang181230) [inline integration/http/src/main/java/io/seata/integration/http/XidResource.java:42]: 这两行其实可以不用改的。
因为bind方法中设置进去的，不可能存在空字符串的，只有null或有值的字符串。

**Pull request merged**